trace-sdk/src/main/java/io/bitrise/trace/data/collector/device/DeviceCarrierDataCollector.java

```diff
@@ -34,7 +36,8 @@ public DeviceCarrierDataCollector(@NonNull final Context context) {
     @Override
     public Data collectData() {
         final Data data = new Data(this);
-        data.setContent(getDeviceCarrier(context));
+        data.setContent(getDeviceCarrier((TelephonyManager) context.getSystemService(
```

Comment:
Why did you move this code from the method below?

Reply:
It was easier to provide a mocked TelephonyManager to the method for the unit tests, than to mock giving it to the context to then get used. If that makes sense?

Reply:
Cool, thanks for the explanation.